### PR TITLE
Implement POST on Session resource

### LINKED ISSUE #165: Implement POST on Session resource
As oneview-redfish-toolkit user, I access "/redfish/v1/SessionService/Sessions" with a post request with oneview credentials on the body according to [the specification](http://redfish.dmtf.org/schemas/DSP0266_1.1.html#session-login). This credentials will be used, together with the value of `[oneview_config]/ip` from redfish.conf file to instantiate a session using python-hpOneview. If the login succeeds, then a response containing a `x-auth-token` generated by OneView should be returned (python-hpOneView maintains the session token inside connection object).

### PR DESCRIPTION
**Pull request by llvieira**
Closes #165.

### CODE REVIEW

File: oneview_redfish_toolkit/api/session.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    import RedfishJsonValidator
+
+
+class Session(RedfishJsonValidator):
+    """Super class of Session Service
+
+        Populates self.redfish with Session response.
+    """
+
+    SCHEMA_NAME = None
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/api/session.py]: There's a session schema. Why are you not validating it?

File: oneview_redfish_toolkit/blueprints/session.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            mimetype='application/json')
+        response.headers.add(
+            "Location", "/redfish/v1/SessionService/Sessions/1")
+        response.headers.add("X-Auth-Token", session_id)
+
+        return response
+
+    except HPOneViewException as e:
+        logging.exception('Unexpected error: {}'.format(e))
+        abort(status.HTTP_400_BAD_REQUEST)
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/blueprints/session.py]: it does not make sense to me to send a bad request on a login failure. Please see if there's a better response

File: oneview_redfish_toolkit/blueprints/session.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    try:
+        try:
+            body = request.get_json()
+            username = body["UserName"]
+            password = body["Password"]
+        except Exception:
+            raise OneViewRedfishError(
+                {"errorCode": "INVALID_INFORMATION",
+                 "message": "Invalid JSON key"})
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/blueprints/session.py]: This message is going to be sent back to user, I would not understand from this message that I created a wrong post form. Please try to create a message that tell to user what happened so he can figure out how to fix it.

File: oneview_redfish_toolkit/api/session.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        super().__init__(self.SCHEMA_NAME)
+
+        self.redfish["@odata.context"] = \
+            "/redfish/v1/$metadata#Session.Session"
+        self.redfish["@odata.id"] = "/redfish/v1/SessionService/Sessions/1"
+        self.redfish["odata.type"] = "#Session.v1_0_0.Session"
+        self.redfish["Id"] = "1"
+        self.redfish["Name"] = "User Session"
+        self.redfish["Description"] = "User Session"
+        self.redfish["UserName"] = username
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/api/session.py]: and add a validation call to the end

File: oneview_redfish_toolkit/tests/blueprints/test_session.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        # Gets json from response
+        json_str = response.data.decode("utf-8")
+
+        # Tests response
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual("application/json", response.mimetype)
+        self.assertEqual(session_mockup, json_str)
+        self.assertIn("/redfish/v1/SessionService/Sessions/1",
+                      response.headers["Location"])
+        self.assertEqual("sessionId", response.headers["X-Auth-Token"])
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/tests/blueprints/test_session.py]: sessionId is hardcoded to 1, how is it matching X-Auth-token and the test is succeeding??

File: oneview_redfish_toolkit/tests/blueprints/test_session.py [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                                     InvalidKey="administrator",
+                                     Password="password")),
+                                 content_type='application/json')
+
+        # Gets json from response
+        json_str = response.data.decode("utf-8")
+
+        with open(
+                'oneview_redfish_toolkit/mockups_errors/'
+                'InvalidJsonKey.json'
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/tests/blueprints/test_session.py]: cant see this file here!!

File: oneview_redfish_toolkit/tests/blueprints/test_session.py [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+        self.assertEqual(json_str, invalid_json_key)
+
+    @mock.patch.object(session_file, 'OneViewClient')
+    def test_post_session_oneview_exception(self, oneview_client_mockup):
+        """Tests post session with HPOneViewException"""
+
+        oneview_client = oneview_client_mockup()
+
+        e = HPOneViewException({
+            'errorCode': 'HTTP_400_BAD_REQUEST',
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/tests/blueprints/test_session.py]: this dont make sense to me as said before

File: oneview_redfish_toolkit/tests/blueprints/test_session.py [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+        # POST Session
+        response = self.app.post("/redfish/v1/SessionService/Sessions",
+                                 data=json.dumps(dict(
+                                     UserName="administrator",
+                                     Password="password")),
+                                 content_type='application/json')
+
+        self.assertEqual(
+            status.HTTP_400_BAD_REQUEST,
```
Comment (ffdarkpenguin) [inline oneview_redfish_toolkit/tests/blueprints/test_session.py]: dont make sense...

File: oneview_redfish_toolkit/tests/blueprints/test_session.py (hunk 6 above)
Comment (llvieira) [reply oneview_redfish_toolkit/tests/blueprints/test_session.py]: https://github.com/HewlettPackard/oneview-redfish-toolkit/blob/master/oneview_redfish_toolkit/mockups_errors/InvalidJsonKey.json

File: oneview_redfish_toolkit/api/session.py (hunk 1 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/session.py]: Fixed.

File: oneview_redfish_toolkit/api/session.py (hunk 4 above)
Comment (llvieira) [reply oneview_redfish_toolkit/api/session.py]: Fixed.

File: oneview_redfish_toolkit/blueprints/session.py (hunk 3 above)
Comment (llvieira) [reply oneview_redfish_toolkit/blueprints/session.py]: Ok, fixed.

File: oneview_redfish_toolkit/blueprints/session.py (hunk 2 above)
Comment (llvieira) [reply oneview_redfish_toolkit/blueprints/session.py]: Changed to `HTTP_401_UNAUTHORIZED`.

File: oneview_redfish_toolkit/tests/blueprints/test_session.py (hunk 7 above)
Comment (llvieira) [reply oneview_redfish_toolkit/tests/blueprints/test_session.py]: Changed to `HTTP_401_UNAUTHORIZED`.

File: oneview_redfish_toolkit/tests/blueprints/test_session.py (hunk 8 above)
Comment (llvieira) [reply oneview_redfish_toolkit/tests/blueprints/test_session.py]: Changed to `HTTP_401_UNAUTHORIZED`.

**Pull request merged**